iocore/hostdb/P_RefCountCache.h

```diff
@@ -0,0 +1,896 @@
+                       (int)refcountcache_current_size_stat, RecRawStatSyncCount);
+
+    RecRegisterRawStat(this->rsb, RECT_PROCESS, (metrics_prefix + "total_inserts").c_str(), RECD_INT, RECP_NON_PERSISTENT,
+                       (int)refcountcache_total_inserts_stat, RecRawStatSyncCount);
+
+    RecRegisterRawStat(this->rsb, RECT_PROCESS, (metrics_prefix + "total_failed_inserts").c_str(), RECD_INT, RECP_NON_PERSISTENT,
+                       (int)refcountcache_total_failed_inserts_stat, RecRawStatSyncCount);
+
+    RecRegisterRawStat(this->rsb, RECT_PROCESS, (metrics_prefix + "total_lookups").c_str(), RECD_INT, RECP_NON_PERSISTENT,
+                       (int)refcountcache_total_lookups_stat, RecRawStatSyncCount);
+
+    RecRegisterRawStat(this->rsb, RECT_PROCESS, (metrics_prefix + "total_hits").c_str(), RECD_INT, RECP_NON_PERSISTENT,
+                       (int)refcountcache_total_hits_stat, RecRawStatSyncCount);
+
+    RecRegisterRawStat(this->rsb, RECT_PROCESS, (metrics_prefix + "last_sync.time").c_str(), RECD_INT, RECP_NON_PERSISTENT,
+                       (int)refcountcache_last_sync_time, RecRawStatSyncCount);
+
+    RecRegisterRawStat(this->rsb, RECT_PROCESS, (metrics_prefix + "last_sync.total_items").c_str(), RECD_INT, RECP_NON_PERSISTENT,
+                       (int)refcountcache_last_total_items, RecRawStatSyncCount);
+
+    RecRegisterRawStat(this->rsb, RECT_PROCESS, (metrics_prefix + "last_sync.total_size").c_str(), RECD_INT, RECP_NON_PERSISTENT,
+                       (int)refcountcache_last_total_size, RecRawStatSyncCount);
+  }
+  // Now lets create all the partitions
+  this->partitions.reserve(num_partitions);
+  for (unsigned int i = 0; i < num_partitions; i++) {
+    this->partitions.push_back(new RefCountCachePartition<C>(i, size / num_partitions, items / num_partitions, this->rsb));
+  }
+}
+
+// Deconstruct the class
+template <class C> RefCountCache<C>::~RefCountCache()
+{
+  delete this->partitions;
+}
+
+template <class C>
+Ptr<C>
+RefCountCache<C>::get(uint64_t key)
+{
+  return this->partitions[this->partition_for_key(key)]->get(key);
+}
+
+template <class C>
+void
+RefCountCache<C>::put(uint64_t key, C *item, int size, int expiry_time)
+{
+  return this->partitions[this->partition_for_key(key)]->put(key, item, size, expiry_time);
+}
+
+// Pick a partition for a given item
+template <class C>
+int
+RefCountCache<C>::partition_for_key(uint64_t key)
+{
+  return key % this->num_partitions;
+}
+
+template <class C>
+RefCountCacheHeader &
+RefCountCache<C>::get_header()
+{
+  return this->header;
+}
+
+template <class C>
+ProxyMutex *
+RefCountCache<C>::lock_for_key(uint64_t key)
+{
+  return this->partitions[this->partition_for_key(key)]->lock.get();
+}
+
+template <class C>
+RefCountCachePartition<C> &
+RefCountCache<C>::get_partition(int pnum)
+{
+  return *this->partitions[pnum];
+}
+
+template <class C>
+const size_t
+RefCountCache<C>::count()
```

Comment:
Why do more ``RefCountCache`` methods appear at the end of the file here? Keep the classes together!